rust/src/datatypes.rs

```diff
@@ -75,10 +79,15 @@ impl Schema {
             .enumerate()
             .find(|&(_, c)| c.name == name)
     }
+}
 
-    pub fn to_string(&self) -> String {
-        let s: Vec<String> = self.columns.iter().map(|c| c.to_string()).collect();
-        s.join(",")
+impl fmt::Display for Schema {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(&self.columns
+            .iter()
+            .map(|c| c.to_string())
+            .collect::<Vec<String>>()
+            .join(", "))
```

Comment:
@sadikovi what do you think of this? I will revert to your suggestion if you prefer. This is nicer from a functional POV

Reply:
Looks good, I suggest you keep your version.